internal/controller/stages/stages.go

```diff
@@ -1005,8 +1005,32 @@ func (r *reconciler) syncPromotions(
 	// current state of the Stage.
 	slices.SortFunc(promotions, kargoapi.ComparePromotionByPhaseAndCreationTime)
 
+	// The latest Promotion is the first element in the sorted list.
+	latestPromo := promotions[0]
+
+	// If the latest Promotion does not match the current Promotion, or is in a
+	// terminal phase, then the current Promotion is no longer valid.
+	if curPromotion := status.CurrentPromotion; curPromotion != nil {
+		if curPromotion.Name != latestPromo.Name || latestPromo.Status.Phase.IsTerminal() {
+			status.CurrentPromotion = nil
+		}
+	}
+
+	// If there is any ongoing verification, we need to let it finish before we
+	// can continue with acknowledging the new Promotion.
+	if curFreightCol := status.FreightHistory.Current(); curFreightCol != nil {
+		for _, verification := range curFreightCol.VerificationHistory {
```

Comment:
Would it make sense to jump right to checking the last one instead of iterating over them all?

Reply:
The reason I picked this approach is because it makes it independent from an assumption that two verification attempts can't be running at the same time.

As the length of the list is limited to 10 items anyway, I think the cost of the loop is negligible.